docs/guides/web/nginx-mainline.md

```diff
@@ -0,0 +1,282 @@
+
+```bash
+sudo mkdir /home/www/
+```
+
+Then, we create the user:
+
+```bash
+sudo adduser -g 'Nginx www user' -h /home/www/ www
+```
+
+That command tells the machine to
+
+* Make a user called “www” (as per the last bit of text), 
+* put all of its files in `/home/www`,
+* and add it to the following groups: “Nginx”, “www”, and “user”.
+
+All three are important, but that “Nginx” group does some real magic. It allows the web server to read and modify files that belong to the “www” user, and the “www” user group. See the Rocky Linux [guide to user management](../../books/admin_guide/06-users.md) for more information.
+
+You will, at this point, be prompted to give the new user a password. Type it in, press Enter, and repeat.
+
+From now on, when you’re actually going to add files to your website, it’s a good idea to do it as the web server user. You can log in to the server user account with by running the following command, and then typing in that password you chose:
+
+```bash
+sudo su www
+```
+
+### Changing the Server Root Folder
+
+Now that you have your fancy new user account, it’s time to make *Nginx* look for your website files in that folder. It;s time to grab your favorite text editor again.
+
+For now, just run:
+
+```bash
+sudo nano /etc/nginx/conf.d
+```
+
+When the file is open, look for the line that looks like `root   /usr/share/nginx/html;`. Change it to your chosen website root folder, eg. `root   /home/www;`. Save and close the file, then test your *Nginx* configuration to make sure you didn’t skip a semi-colon or anything:
+
+```bash
+nginx -t
+```
+
+If you get the collowing success message, everything went right:
+
+```
+nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
+nginx: configuration file /etc/nginx/nginx.conf test is successful
+```
+
+Then, give the server a soft restart with:
+
+```bash
+sudo systemctl reload nginx
+```
+
+If the soft restart doesn’t work, give *Nginx* a kick in the pants with:
+
+```bash
+sudo systemctl restart nginx
+```
+
+Any HTML files in your new root folder should now be browsable from… your browser.
+
+### Changing File Permissions
+
+To make sure that *Nginx* can read, write to, and execute any files in the website directory, permissions need to be set properly, especially if you uploaded the files while using the root account. 
+
+First, make sure that all files in the root folder are owned by the server user and its user group with:
+
+```bash
+sudo chown -R www:www /home/www
+```
+
+And then, to make sure that users who want to actually browse your website can actually see the pages, you should make you can run this command:
+
+```bash
+sudo chmod -R 755 /home/www 
+```
+
+That basically gives everyone the right to look at files on the server, but not modify them. Only the owners get to do that. If you’re feeling paranoid, you can run that particular command every time you upload new HTML/CSS/JS/image files to a static website.
+
```

Comment:
For a better security, only files or directory that you want to be writeable should belongs to the dedicated user. For the rest of the files, readonly (o=r) is sufficient to be served by the webserver.

Reply:
Fair enough. Time to explain more again, I guess.